src/main/java/org/apache/xml/security/encryption/XMLCipher.java

```diff
@@ -1364,26 +1377,41 @@ public EncryptedKey encryptKey(
         } else {
             c = contextCipher;
         }
-        // Now perform the encryption
 
+        AlgorithmParameterSpec chiperSpec = null;
+        KeyPair origninatorKeyPair = null;
+        Key wrapKey = this.key;
+        if (params instanceof OAEPParameterSpec) {
+            chiperSpec = params;
+            wrapKey = this.key;
+        } else if (params instanceof KeyAgreementParameterSpec) {
+            KeyAgreementParameterSpec keyAgreementParameter = (KeyAgreementParameterSpec) params;
+            PublicKey recipientPublicKey = (PublicKey) this.key;
+            if (keyAgreementParameter.getRecipientPublicKey() == null) {
+                keyAgreementParameter.setRecipientPublicKey(recipientPublicKey);
+            }
+            if (keyAgreementParameter.getOriginatorPrivateKey() == null) {
+                origninatorKeyPair = KeyUtils.generateEphemeralDHKeyPair(recipientPublicKey, null);
+                keyAgreementParameter.setOriginatorKeyPair(origninatorKeyPair);
+            }
+
+            wrapKey = KeyUtils.aesWrapKeyWithDHGeneratedKey(keyAgreementParameter);
+        }
```

Comment:
Should throw an exception if params are unrecognized.

Reply:
Done